nifi-commons/nifi-utils/src/main/java/org/apache/nifi/util/file/classloader/ClassLoaderUtils.java

```diff
@@ -21,45 +21,59 @@
 import java.net.MalformedURLException;
 import java.net.URL;
 import java.net.URLClassLoader;
+import java.util.Arrays;
 import java.util.LinkedList;
 import java.util.List;
+import java.util.stream.Collectors;
 
 public class ClassLoaderUtils {
 
     public static ClassLoader getCustomClassLoader(String modulePath, ClassLoader parentClassLoader, FilenameFilter filenameFilter) throws MalformedURLException {
-        String[] modules = modulePath != null? modulePath.split(",") : null;
-        URL[] classpaths = getURLsForClasspath(modules,filenameFilter);
-        return createModuleClassLoader(classpaths,parentClassLoader);
+        // Split and trim the module path(s)
+        List<String> modules = (modulePath == null)
+                ? null
+                : Arrays.stream(modulePath.split(",")).map((path) -> path == null ? null : path.trim()).filter((item) -> item != null).collect(Collectors.toList());
```

Comment:
Executed tests and realized doesn't quite cover for mistakenly added leading commas or leading/trailing commas with blanks before or after them (respectively) e.g. "/src/folder/one.jar1, /src/folder/two.jar, " . Edge case but I think changing the lambda around to blank values in array are filtered out should prevent error:

//Maybe something like this using StringUtils.isNotBlank ?
Arrays.stream(modulePath.split(",")).filter((item) -> StringUtils.isNotBlank(item)).map((path) -> path.trim()).collect(Collectors.toList());